### PR TITLE
Add READMEs for search jobs

### PR DESCRIPTION
**Pull request by loic-sharma**
Addresses https://github.com/nuget/nugetgallery/issues/8005

### CODE REVIEW

File: src/NuGet.Jobs.Auxiliary2AzureSearch/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+## Overview
+
+**Subsystem: Search 🔎**
+
+This job updates the [search auxiliary files](../../docs/Search-auxiliary-files.md) used by the [search service](../NuGet.Services.SearchService). This also updates the downloads data in the [Azure Search "search" index](../../docs/Azure-Search-indexes.md).
```
Comment (joelverhagen) [inline src/NuGet.Jobs.Auxiliary2AzureSearch/README.md:5]: I would say it updates the downloads **and** owners data

File: src/NuGet.Jobs.Catalog2AzureSearch/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+NuGet.Jobs.Catalog2AzureSearch.exe -Configuration path\to\your\settings.json
+```
+
+### Using DEV resources
+
+The easiest way to run the tool if you are on the nuget.org team is to use the DEV environment resources:
+
+1. Install the certificate used to authenticate as our client AAD app registration into your `CurrentUser` certificate store.
+1. Clone our internal [`NuGetDeployment`](https://nuget.visualstudio.com/DefaultCollection/NuGetMicrosoft/_git/NuGetDeploymentp) repository.
+1. Update your cloned copy of the [DEV Catalog2AzureSearch appsettings.json](https://nuget.visualstudio.com/DefaultCollection/NuGetMicrosoft/_git/NuGetDeployment?path=%2Fsrc%2FJobs%2FNuGet.Jobs.Cloud%2FJobs%2FCatalog2AzureSearch%2FDEV%2Fnorthcentralus%2Fa%2Fappsettings.json) file to authenticate using the certificate you installed:
```
Comment (joelverhagen) [inline src/NuGet.Jobs.Catalog2AzureSearch/README.md:23]: I like it. Simple.

File: src/NuGet.Jobs.Db2AzureSearch/README.md [hunk 3]
```diff
@@ -1 +1,122 @@
...
+### Using personal Azure resources
+
+As an alternative to using nuget.org's DEV resources, you can also run this tool using your personal Azure resources.
+
+#### Prerequisites
+
+- **Gallery DB**. This can be initialized locally using the [NuGetGallery](https://github.com/NuGet/NuGetGallery/blob/master/README.md).
+- **Azure Search**. You can create your own Azure Search resource using the [Azure Portal](https://docs.microsoft.com/en-us/azure/search/search-create-service-portal).
+- **Azure Blob Storage**. You can create your own Azure Blob Storage using the [Azure Portal](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create). You will need to create a container named `ng-search-data` and upload the following files:
+    1. [`downloads.v1.json`](https://nugetprod0.blob.core.windows.net/ng-search-data/downloads.v1.json)
```
Comment (joelverhagen) [inline src/NuGet.Jobs.Db2AzureSearch/README.md:52]: We should not document the explicit URLs but rather how to discover them from nugat deployment repo.

File: src/NuGet.Jobs.Auxiliary2AzureSearch/README.md (hunk 1 above)
Comment (loic-sharma) [reply src/NuGet.Jobs.Auxiliary2AzureSearch/README.md:5]: Oops good catch

Comment (loic-sharma) [reply src/NuGet.Jobs.Auxiliary2AzureSearch/README.md]: Fixed

File: src/NuGet.Jobs.Db2AzureSearch/README.md (hunk 3 above)
Comment (loic-sharma) [reply src/NuGet.Jobs.Db2AzureSearch/README.md]: Fixed. I also added "empty" content descriptions for that developers that aren't on the nuget.org team

File: docs/Search-version-list-resource.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+**Subsystem: Search 🔎**
+
+The version list resource is bookkeeping used to update the search subsytem. Effectively, this resource is a mapping of package IDs to their versions. It is initially created by the [Db2AzureSearch](../src/NuGet.Jobs.Db2AzureSearch) tool and is kept up-to-date by the [Catalog2AzureSearch](../src/NuGet.Jobs.Catalog2AzureSearch) job.
+
+## Purpose
+
+The [search index](./Azure-Search-indexes.md#Search-index) stores up to 4 documents per package ID. These documents store the metadata for the latest listed version across these pivots:
+
+1. Prerelease - Includes packages that aren't stable
+1. SemVer - Includes packages that require SemVer 2.0.0 support
```
Comment (t-mog-msft) [inline docs/Search-version-list-resource.md:12]: Did you mean "2." here?

File: src/NuGet.Jobs.Auxiliary2AzureSearch/README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+This job is a singleton. Only a single instance of the job should be running per Azure Search resource.
+
+### Using DEV resources
+
+The easiest way to run the tool if you are on the nuget.org team is to use the DEV environment resources:
+
+1. Install the certificate used to authenticate as our client AAD app registration into your `CurrentUser` certificate store.
+1. Clone our internal [`NuGetDeployment`](https://nuget.visualstudio.com/DefaultCollection/NuGetMicrosoft/_git/NuGetDeploymentp) repository.
+1. Update your cloned copy of the [DEV Auxiliary2AzureSearch appsettings.json](https://nuget.visualstudio.com/DefaultCollection/NuGetMicrosoft/_git/NuGetDeployment?path=%2Fsrc%2FJobs%2FNuGet.Jobs.Cloud%2FJobs%2FAuxiliary2AzureSearch%2FDEV%2Fnorthcentralus%2Fa%2Fappsettings.json) file to authenticate using the certificate you installed:
```
Comment (t-mog-msft) [inline src/NuGet.Jobs.Auxiliary2AzureSearch/README.md:23]: 1.  2. 3. ?

File: src/NuGet.Jobs.Auxiliary2AzureSearch/README.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    1. Determine which packages have popularity transfer changes
+    1. Update Azure Search documents in the "search" index to reflect the latest downloads and popularity transfers
+    1. Save the "new" downloads data to the search auxiliary storage
+    1. Save the "new" popularity transfers data to search auxiliary storage
+1. Update owners
+    1. Get the "old" owners data from search auxiliary storage
+    1. Get the "new" owners data from Gallery DB
+    1. Update Azure Search documents in the "search" index to reflect the ownership changes, if any
+    1. Track ownership changes in search auxiliary storage
+    1. Save the "new" owners data to the search auxiliary storage
```
Comment (t-mog-msft) [inline src/NuGet.Jobs.Auxiliary2AzureSearch/README.md:112]: Bunch of ones here. Should be 1, 2, 3, etc.

File: docs/Search-version-list-resource.md (hunk 4 above)
Comment (loic-sharma) [reply docs/Search-version-list-resource.md:12]: Markdown properly renders this: https://github.com/NuGet/NuGet.Services.Metadata/blob/b0a1dc0c2f6124be498845fb8e3c794da78a6af9/src/NuGet.Jobs.Auxiliary2AzureSearch/README.md#using-dev-resources

Using `1.` repeatedly lets you move items in the numbered list without needing to update its number. This trick makes it a little easier to maintain READMEs, but it does make the "raw" markdown a tiny bit less readable. I'm ok with this tradeoff given that most people will read this README on GitHub.

File: src/NuGet.Services.SearchService/README.md [hunk 7]
```diff
@@ -356,5 +356,9 @@ The `ApplicationInsights_InstrumentationKey` setting is optional and can be remo
 
 ## Prerequisites
 
+TODO: Point this to Db2AzureSearch.
```
Comment (t-mog-msft) [inline src/NuGet.Services.SearchService/README.md:359]: This is already done no?

File: src/NuGet.Jobs.Auxiliary2AzureSearch/README.md (hunk 5 above)
Comment (loic-sharma) [reply src/NuGet.Jobs.Auxiliary2AzureSearch/README.md:23]: Yup see my comment here: https://github.com/NuGet/NuGet.Services.Metadata/pull/797#discussion_r449178372

File: src/NuGet.Jobs.Auxiliary2AzureSearch/README.md (hunk 6 above)
Comment (loic-sharma) [reply src/NuGet.Jobs.Auxiliary2AzureSearch/README.md:112]: Yup see my comment here: https://github.com/NuGet/NuGet.Services.Metadata/pull/797#discussion_r449178372

File: src/NuGet.Services.SearchService/README.md (hunk 7 above)
Comment (loic-sharma) [reply src/NuGet.Services.SearchService/README.md:359]: Oops, good catch!

Comment (loic-sharma) [reply src/NuGet.Services.SearchService/README.md]: Fixed

File: docs/Search-version-list-resource.md (hunk 4 above)
Comment (t-mog-msft) [reply docs/Search-version-list-resource.md:12]: Yeah I get the maintenance point here and agree that most people will read this on GitHub where it's correctly rendered 😃

**Pull request merged**